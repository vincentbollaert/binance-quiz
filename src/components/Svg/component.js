import React, { memo } from 'react'
import { string, number, func } from 'prop-types'
import styled from 'styled-components'

const Wrap = styled.span`
  display: flex;
  flex-shrink: 0;
  width: ${props => props.size || props.width}rem;
  height: ${props => props.size || props.height}rem;

  svg {
    width: 100%;
    height: 100%;
  };
`

const Svg = ({ svg, width, size, className, onClick }) => (
  <Wrap
    width={width}
    size={size}
    className={className}
    onClick={onClick}
    dangerouslySetInnerHTML={{ __html: svg }}
    data-testid="component-svg"
  />
)

Svg.propTypes = {
  svg: string.isRequired,
  width: number,
  size: number,
  onClick: func,
  className: string,
}

Svg.defaultProps = {
  width: undefined,
  size: undefined,
  onClick() {},
  className: undefined,
}

export default memo(Svg)
