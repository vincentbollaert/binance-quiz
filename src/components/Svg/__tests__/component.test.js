import '@testing-library/jest-dom'
import React from 'react'
import { render } from '@testing-library/react'
import Svg from '../component'

describe('Svg', () => {
  const props = {
    size: 2,
    svg: '',
  }
  test('matches snapshot', () => {
    const { container } = render(<Svg {...props} />)
    expect(container.firstChild).toMatchSnapshot()
  })
  test('sets correct styles', () => {
    const { queryByTestId, rerender } = render(<Svg {...props} />)
    expect(queryByTestId('component-svg')).toHaveStyle({ width: '2rem', height: '2rem' })

    rerender(<Svg svg="" width={1} />)
    expect(queryByTestId('component-svg')).toHaveStyle({ width: '1rem' })
  })
})
