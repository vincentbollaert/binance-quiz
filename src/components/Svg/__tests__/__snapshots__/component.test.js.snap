// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Svg matches snapshot 1`] = `
<span
  class="sc-AxjAm inReEw"
  data-testid="component-svg"
  size="2"
/>
`;
